src/jobs/actions/rabbitmqaction.ts

```diff
@@ -0,0 +1,91 @@
+import { Logger, NotFoundException } from "@nestjs/common";
+import amqp, { Connection } from "amqplib/callback_api";
+import { JobAction } from "../config/jobconfig";
+import { JobClass } from "../schemas/job.schema";
+
+
+/**
+ * Publish a message following a job status update
+ */
+export class RabbitMQJobAction<T> implements JobAction<T> {
+  public static readonly actionType = "rabbitmq";
+  private connectionDetails;
+  private queueName;
+
+  constructor(data: Record<string, any>) {
+    Logger.log(
+      "Initializing RabbitMQJobAction. Params: " + JSON.stringify(data),
+      "RabbitMQJobAction",
+    );
+
+    this.connectionDetails = {
+      protocol: "amqp",
+      hostname: data.hostname,
+      port: data.port,
+      username: data.username,
+      password: data.password,
+    };
+    this.queueName = data.queue;
+  }
+
+  getActionType(): string {
+    return RabbitMQJobAction.actionType;
+  }
+
+  async validate(dto: T) {
+    Logger.log(
+      "Validating RabbitMQJobAction: " + JSON.stringify(dto),
+      "RabbitMQJobAction",
+    );
+
+    if ([undefined, ""].some(el => Object.values(this.connectionDetails).includes(el))) {
+      throw new NotFoundException("Configuration is missing connection details.");
+    }
+    if (this.queueName == undefined || this.queueName == "") {
+      throw new NotFoundException("Queue name is not defined.");
+    }
```

Comment:
I'm not sure about throwing here. Wouldn't it crash the application entirely?

Reply:
Valid point. We should have tests for this and implement the correct behavior.

Reply:
Throwing an error here does not crash the application. If, for example, you want to create a RabbitMQ action when creating a job, but the RabbitMQ details are not defined properly, then you get an error and the job does not get created, but the application continues to run.